src/prefect/engine.py

```diff
@@ -1169,11 +1169,22 @@ async def orchestrate_task_run(
         # Resolve futures in any non-data dependencies to ensure they are ready
         await resolve_inputs(wait_for, return_data=False)
     except UpstreamTaskError as upstream_exc:
-        return await propose_state(
-            client,
-            Pending(name="NotReady", message=str(upstream_exc)),
-            task_run_id=task_run.id,
-        )
+        if task_run.state.is_pending():
+            # if orchestrating a run already in a pending state, force orchestration to
+            # update the state name
+            return await propose_state(
+                client,
+                Pending(name="NotReady", message=str(upstream_exc)),
+                task_run_id=task_run.id,
+                force=True,
```

Comment:
Can we just do `force=task_run.state.is_pending()` with that comment for clarity?